test/clippy/clippy_failure_test.sh

```diff
@@ -77,9 +80,15 @@ EOF
   check_build_result $BUILD_FAILED bad_library_clippy
   check_build_result $BUILD_FAILED bad_test_clippy
 
+  # When capturing output, clippy errors are treated as warnings and the build
+  # should succeed.
+  check_build_result $BUILD_OK bad_binary_clippy True
```

Comment:
Nit: using the local -r to declare CAPTURE_OUTPUT=True and DONT_CAPTURE_OUTPUT=False and using that in the test instead of True/False directly would make the tests more readable IMHO.

I'd also make all arguments to check_build_result mandatory and always passed.

Reply:
I made it so arguments >=3 are just passed to `bazel build` as-is. I also used `local -r`, which I agree improves readability.

The reason I chose not to pass extra args to bazel in the default case is because I think it better reflects how it will be actually be invoked. The vast majority of the time, you won't set `--@rules_rust//:capture_clippy_output`, but will depend on the default value. If you feel strongly I'm happy to change it, though.

Reply:
Even better, awesome!